decred/decred/dcr/wire/msgpong.py

```diff
@@ -0,0 +1,84 @@
+"""
+Copyright (c) 2020, The Decred developers
+See LICENSE for details.
+
+Based on dcrd MsgPing.
+"""
+
+from decred.util.encode import ByteArray
+
+
+CmdPong = "pong"
+
+NonceLength = 8
+
+
+class MsgPong:
+    """
+    MsgPong implements the Message API and represents a Decred pong message
+    which is used primarily to confirm that a connection is still valid in
+    response to a Decred ping message (MsgPing).
+
+    This message was not added until protocol versions AFTER BIP0031Version.
+    """
+
+    def __init__(self, nonce):
+        """
+        Args:
+            nonce (int): Unique value associated with message that is used to
+                identify specific ping message.
+        """
+        self.nonce = nonce
+
+    @staticmethod
+    def btcDecode(b, pver):
+        """
+        btcDecode decodes b using the Decred protocol encoding into the receiver. This
+        is part of the Message API.
+
+        Args:
+            b (ByteArray): The encoded MsgPong.
+            pver (int): The protocol version. Unused.
+
+        Returns:
+            MsgPong. The MsgPong.
+        """
+        return MsgPong(nonce=b.unLittle().int())
+
+    def btcEncode(self, pver):
+        """
+        btcEncode encodes the receiver to w using the Decred protocol encoding.
```

Comment:
What's "w", the wire protocol?

Reply:
Whoops. That's Go thing. Parameter name for an `io.Writer` for writing byte data.